src/providers/tfc/workspaceProvider.ts

```diff
@@ -4,23 +4,35 @@
  */
 
 import * as vscode from 'vscode';
+import axios from 'axios';
+
 import { RunTreeDataProvider } from './runProvider';
 import { apiClient } from '../../terraformCloud';
 import { TerraformCloudAuthenticationProvider } from '../authenticationProvider';
-import axios from 'axios';
+import { ProjectQuickPick } from './workspaceFilters';
 
 export class WorkspaceTreeDataProvider implements vscode.TreeDataProvider<WorkspaceTreeItem>, vscode.Disposable {
   private readonly didChangeTreeData = new vscode.EventEmitter<void | WorkspaceTreeItem>();
   public readonly onDidChangeTreeData = this.didChangeTreeData.event;
 
   constructor(private ctx: vscode.ExtensionContext, private runDataProvider: RunTreeDataProvider) {
     vscode.commands.registerCommand('terraform.cloud.workspaces.refresh', () => this.refresh());
+    vscode.commands.registerCommand('terraform.cloud.workspaces.selectProject', () => this.selectProject());
   }
 
   refresh(): void {
     this.didChangeTreeData.fire();
   }
 
+  async selectProject(): Promise<void> {
+    // TODO! only run this if user is logged in
+    const organization = this.ctx.workspaceState.get('terraform.cloud.organization', '');
+    const projectQuickPick = new ProjectQuickPick(organization);
+    const project = await projectQuickPick.pick();
+    // TODO! use project for filtering
```

Comment:
The UX of this is great! It just leaves off actually using the selected project :(